docs/customization.md

```diff
@@ -1,38 +1,53 @@
 ## Customization
-### Modifying the client library format
 
-Don't like the default client syntax?  Want a different language supported?  No problem!  OpenAPI Generator processes mustache templates with the [jmustache](https://github.com/samskivert/jmustache) engine.  You can modify our templates or make your own.
+### Modifying a template
 
-You can look at `modules/openapi-generator/src/main/resources/${your-language}` for examples. To make your own templates, create your own files and use the `-t` flag to specify your template folder.  It actually is that easy.
+Clone OpenAPI Generator and take a look at the following directory: `modules/openapi-generator/src/main/resources/${template}`. In here you'll see all of the generators available, for most programming languages, web application frameworks and web servers.
 
-### Making your own codegen modules
+Templates consist of multiple mustache files. Mustache is a simple, readable templating language, and the engine used is [jmustache](https://github.com/samskivert/jmustache).
 
-If you're starting a project with a new language and don't see what you need, openapi-generator can help you create a project to generate your own libraries:
+If you wish to modify one of these templates, copy and paste the template you're interested in to a templates directory you control. To let OpenAPI Generator know where this templates directory is, use the `-t` option (e.g: `-t ./templates/`).
+
+To tie that all together:
+
+```sh
+mkdir templates
+cp -r modules/openapi-generator/src/main/resources/${template} templates/
```

Comment:
I don't know if it's helpful to add to this doc or what… but if you install [refined github](https://github.com/sindresorhus/refined-github/) in Google Chrome, you get a nifty `Download` button under every folder as you browse GitHub. I use this often, as you can easily browse to your desired generator that you'd like to extend under `modules/openapi-generator/src/main/resources/` and download the folder directly.

Reply:
I don't wanna push people towards a browser extension when a single command would also explain things just fine.